src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -75,7 +75,10 @@ public SinkIslandCheck(final Configuration configuration)
     public boolean validCheckForObject(final AtlasObject object)
     {
         return this.validEdge(object) && !this.isFlagged(object.getIdentifier()) && ((Edge) object)
-                .highwayTag().isMoreImportantThanOrEqualTo(this.minimumHighwayType);
+                .highwayTag().isMoreImportantThanOrEqualTo(this.minimumHighwayType)
+        // Ignore edges that are fully enclosed in areas with amenity values to exclude
+                && !(this.isServiceRoad((Edge) object)
+                        && this.isWithinAreasWithExcludedAmenityTags((Edge) object));
```

Comment:
Any implications on how running this function for every Edge will affect the runtime ?

Reply:
Great question! I had not analyzed how long it take to run this for every edge. But the previous enhancement was doing the same for every edge and for every connected edges, which we are not doing anymore with this fix.